src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java

```diff
@@ -308,6 +308,18 @@ public boolean exists(@Nonnull F<T, Boolean> f) {
         }
     }
 
+    @Override
+    public boolean every(@Nonnull F<T, Boolean> f) {
+        ImmutableList<T> list = this;
+        while (list instanceof NonEmptyImmutableList) {
```

Comment:
Can we change the implementation to match `exists`? There's no real reason for them to be different.